tests/coverage/branch/guard-simple.coverage

```diff
@@ -0,0 +1,30 @@
+   LL|       |#![feature(coverage_attribute)]
+   LL|       |//@ edition: 2021
+   LL|       |//@ compile-flags: -Zcoverage-options=branch
+   LL|       |//@ llvm-cov-flags: --show-branches=count
+   LL|       |
+   LL|       |use core::hint::black_box;
+   LL|       |
+   LL|      1|fn never_taken() {
+   LL|      1|    match black_box(false) {
+   LL|      1|        _ if black_box(false) => {}
```

Comment:
I am wondering why the pattern `_` and the guard `black_box(false)` are in the same code region, while there are 2 branch expressions.

I would have expected the guard to have its own counter displayed underneath
```
   LL|      1|        _ if black_box(false) => {}
                           ^1                  ^0
```